pandas/core/frame.py

```diff
@@ -3025,10 +3025,14 @@ def _set_value(self, index, col, value, takeable: bool = False):
 
             series = self._get_item_cache(col)
             engine = self.index._engine
-            engine.set_value(series._values, index, value)
+            loc = engine.get_loc(index)
```

Comment:
Any chance of just updating the `set_value` implementation to include this validation?

Reply:
i think we can simplify the code so that the validation is done in fewer places, but i dont think its going to end up being in set_value